### ISSUE TITLE
Migrations queryInterface.addColumn before/after another column?

### ISSUE THREAD

**Issue by micksatana** [sequelize/sequelize#7359]
~~Anyhow can I use queryInterface.addColumn to add a column before or after another specific column?~~
It'd be nice to have this feature.

# Add a column after another column
```js
queryInterface.addColumn('tableA', 'columnC', Sequelize.STRING, {
    after: 'columnB'
});
```

## What do you expect to happen?
The column C should be added after columnB

## What is actually happening?
The column append at the end of table

# Add a column as the first column in the table

```js
queryInterface.addColumn('tableA', 'columnA', Sequelize.STRING, {
    first: true
});
```

## What do you expect to happen?
The columnA should be added as the first column of the table

## What is actually happening?
The column append at the end of table

__Dialect:__ mysql
__Database version:__ MySQL 5.6.20
__Sequelize version:__ Sequelize [Node: 6.7.0, CLI: 2.4.0, ORM: 3.24.3, mysql: ^2.11.1]

**Comment by relwell** [comment]
This one is currently burning me as well, on Postgres.

**Comment by daniel-huckins** [comment]
doesn't look like the underlying supports 'options', any of the official maintainers have an objection to adding a parameter so 'before' and 'after' are possible?
https://github.com/sequelize/sequelize/blob/master/lib/dialects/mysql/query-generator.js#L102

**Comment by micksatana** [comment]
Thank you Daniel. I'll edit as requested feature instead.

**Comment by micksatana** [comment]
According to MySQL syntax, there is no BEFORE keyword for ADD COLUMN so this feature should use the keywords FIRST and AFTER instead. i'll edit the issue description.

**Comment by micksatana** [comment]
@janmeier can you help re-open? I finish coding the feature for MySQL with unit tests but can't send a pull request.

**Comment by micksatana** [comment]
@janmeier Thank you for fast response. I still can't send a pull request though. I send using GitHub Desktop, it shows "Authentication Failed"

This is my first pull request ever in GitHub so really sorry for stupid questions. Do I need to get permission first before sending a pull request? If so, who can help set the permission for me?

**Comment by micksatana** [comment]
Already merged, thank you.

**Comment by dhuckins** [comment]
You might need to fork it and then make a PR from the website

**Comment by abdullah2993** [comment]
How about postgres support?

**Comment by JefferyHus** [comment]
@abdullah2993 postgres doesn't support adding a column AFTER another one.

**Comment by sw-tt-bansariakhani** [comment]
Do we have this option for mssql ?

**Comment by ephys** [comment]
mssql doesn't support specifying the order of added columns in ALTER TABLE. Afaik only mysql supports this.
